functions.lua

```diff
@@ -117,11 +117,9 @@ xp_redo.add_xp = function(playername, xp)
 	if currentRank and currentRank.xp > previousRank.xp then
 		-- level up
 		xp_redo.run_hook("rank_change", { playername, sumXp, currentRank })
-
-		local state = player:get_attribute(xp_redo.HUD_DISPLAY_STATE_NAME)
-		if state and state == "on" then
-			level_up(player, currentRank)
-		end
+		level_up(player, currentRank)
+		local text = "Congratulations, "..playername.." reached "..currentRank.name
+		minetest.chat_send_all(minetest.colorize("#98ff98",text))
```

Comment:
This should be configurable, that is allow disabling global messages.

Reply:
Most of the other stuff should also be configurable, right?

I mean, not everyone wants players to lose a fixed 1000 XP as it is now, but maybe I should make that configurable too. Maybe even the fixed amount or the % of the next level that gets lost on death. depending on the level formula, 10% of the next level might be much more than those 1000 XP.

Same goes for not losing levels. I need that for my game. But someone else's game may depend on people losing levels.

The message also needs to be translatable.

Reply:
> Most of the other stuff should also be configurable, right?

Yes of course it would be nice but I don't think that belongs to this pull request.

> not everyone wants players to lose a fixed 1000 XP as it is now, but maybe I should make that configurable too.

That would be nice too, keep that 1000 as default and add configuration option, however again not that important for this PR.

> The message also needs to be translatable.

That would be nice, actual translations of course can be added later too. Actually I have no idea how translations work in chat messages... if client side translation even works there. If it does not then it gets bit complicated.

Only thing I see as important so far for this exact PR before merging would be option to disable chat message, later I could also see good configuration option to partially disable as in only send level up messages to everyone if player has say > 20k XP.

Reply:
Actually seems that I did misread whole thing, yes XP reduction type should be configurable and possibly overrideable, select from few basic: default -1000 or optionally percentage. There could be 2 config options: type and amount. Keep current behavior as default.

Ranks, not sure how important this is. It might be but I suppose most would track XP amount but could be wrong...